scripts/merge-analyses.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+
+# This script merges the analysis data in the files provided as
+# arguments, and prints the merged analysis data to stdout.
+# The "target" data lines from the input files are copied to
+# the output as-is. The "source" data lines are merged such
+# that the syntax and symbol properties are unioned across
+# all input lines that have a matching (loc, pretty) tuple.
+# This ensures that for a given identifier, only a single context
+# menu item will be displayed for a given "pretty" representation,
+# and that context menu will link to all the symbols from all
+# the input files that match that.
+
+# Note that this code intentionally preserves the order of stuff
+# within each line, as much as possible. However, the lines
+# themselves get reordered, but the output can be piped through
+# `sort` as needed.
+
+from collections import OrderedDict
+import json
+import sys
+
+sourcemap = {}
+
+def SplitToOrderedSet(comma_sep_str):
+    return OrderedDict([(item, None) for item in comma_sep_str.split(',')])
+
+for filename in sys.argv[1:]:
+    filehandle = open(filename)
+    for line in filehandle.readlines():
+        # parse the JSON into an OrderedDict to maintain order of properties
+        obj = json.loads(line, object_pairs_hook=OrderedDict)
+
+        if "source" not in obj:
+            print(line.strip())
+            continue
+
+        key = obj["loc"] + obj["pretty"]
+        if key in sourcemap:
+            # We already encountered a line with this (loc,pretty) tuple,
+            # so merge the syntax and sym properties. Again use an
+            # OrderedDict (with None values, so it is effectively an
+            # ordered set) to avoid reordering the existing tokens.
+            entry = sourcemap[key]
```

Comment:
Maybe assert that there are no other unexpected fields in |entry|?

Reply:
Will do